T683 delete project modal confirm

OndraZizka opened the pull request:
I don't get how <foo #something >works in Angular2. The docs say I can refer to it anywhere in the template. When I do, it's not defined. See the modal component's #confirmInput and {{confirmInput.value}}.

klinki commented on ui/src/main/webapp/src/app/project/project-list.component.html:
```diff
@@ -33,16 +33,15 @@ <h1 i18n="Project page header">Projects</h1>
             <div *ngFor="let project of projects"
-                class="list-group-item  project-info  tile-click"
-                (click)="viewProject($event, project)">
+                class="list-group-item  project-info  tile-click">
```
You removed click event, but that is wrong - clicking on tile must work.

OndraZizka replied:
Right, that's because otherwise any click resulted in navigation event

OndraZizka replied:
Fixed

klinki commented on ui/src/main/webapp/src/app/project/project-list.component.html:
```diff
@@ -80,6 +90,6 @@ <h1 i18n="Project page header">Projects</h1>
                 (confirmed)="doDeleteProject(project)"
         >
         </wu-confirmation-modal>
```
Have you tried to use solution with just 1 modal dialog? It doesn't work?

OndraZizka replied:
I haven't, would take more time which we don't have.

klinki commented on ui/src/main/webapp/src/app/project/project-list.component.html:
```diff
@@ -33,16 +33,15 @@ <h1 i18n="Project page header">Projects</h1>
                     </span>
                     <span class="fa-lg project-action-item" i18n-title="Delete project" title="Delete project">
-                        <a (click)="deleteProject($event, project)"  class="action-button">
+                        <a (click)="$event.stopPropagation(); deleteProjectModal.show($event, project)"  class="action-button">
```
Why you don't use `confirmDeleteProject` method?

Btw you, who criticize having too much logic in templates, are putting this `$event.stopPropagation` here? :)

OndraZizka replied:
Well, I admit I could

OndraZizka replied:
Let's do it after summit

klinki replied:
What? It would take 5 minutes to do that properly, why after summit?

klinki commented on ui/src/main/webapp/src/app/shared/confirmation-modal.component.ts:
```diff
@@ -16,11 +16,16 @@ import * as $ from "jquery";
+                <p>{{body}}</p>
+                <p [style.display]="confirmPhrase ? 'block' : 'none'">
+                    <input #confirmInput style="margin: 1ex 1em" type="text"
+                       (keyup)="yes.disabled = (confirmPhrase && ($event.target.value.trim().toLowerCase() != confirmPhrase.trim().toLowerCase()))" >
```
I would prefer having this code in class instead of template.
(I'm quite surprised you put it here)

OndraZizka replied:
Ok, let's fix that after summit.

OndraZizka commented:
you should keep current modal dialog (as we will probably use it in some other places - for example deleting applications) and create a new one, which will require typing name for confirmation

Why not reuse it? The changes are conditional based on confirmPhrase.

    It doesn't really work - I didn't type anything and it allowed me to click 'Delete the project' anyway
    when I type something, delete button becomes inactive (maybe you accidentally switched condition?)

That was the "before 1st keyup event" thing. fixed.

    It should show name of the project in modal dialog (as OpenShift does)

Fixed

    When you click cancel and open dialog again, it stays filled - state must be reset

This is highly improbable scenario :) But if it's worth few minutes figuring out how, then why not

    I'm pretty sure Marc wouldn't be happy with UI - textbox feels too small

Did he have any specific wishes on this? How big should it be?

    I recommend you to look at OpenShift and copy as much of their UI as possible

Ok I'll try

klinki commented:
> 
>     you should keep current modal dialog (as we will probably use it in some other places - for example deleting applications) and create a new one, which will require typing name for confirmation
> 
> Why not reuse it? The changes are conditional based on confirmPhrase.

Ok, I agree. I didn't test it in other case, but if it works and doesn't cause any problems, then it is acceptable for me.


>     When you click cancel and open dialog again, it stays filled - state must be reset
> 
> This is highly improbable scenario :) But if it's worth few minutes figuring out how, then why not

I cannot agree with that. This must be fixed before merging.


>     I'm pretty sure Marc wouldn't be happy with UI - textbox feels too small
> 
> Did he have any specific wishes on this? How big should it be?
> 
>     I recommend you to look at OpenShift and copy as much of their UI as possible
> 
> Ok I'll try

I don't really care that much about UI, but I'm pretty sure Marc would have problems with it if it doesn't look good enough. So looking at OpenShift is just a recommendation. It shouldn't be wrong, if it looks like OpenShift.

Just one point here - if button is enabled, it should have different style. It should be blue, as we have it everywhere else.
